### PR TITLE
Improve resource descriptions for documentation

### PR DESCRIPTION
**Pull request by tas50**
Mostly cleaning up how we describe name properties, but also copying over some edits that happened on the docs site.

Signed-off-by: Tim Smith <tsmith@chef.io>

### CODE REVIEW

**Comment by tas50** [comment]
Ping @chef/docs-team as this syncs up the changes that were recently made to the docs site

File: lib/chef/resource/cron_d.rb [hunk 1]
```diff
@@ -87,7 +87,7 @@ def self.validate_dow(spec)
       end
 
       property :cron_name, String,
-               description: "Set the name of the cron job. If this isn't specified we'll use the resource name.",
+               description: "An optional property to set the cron name if it differs from the resource block's name.",
```
Comment (tyler-ball) [inline lib/chef/resource/cron_d.rb]: `cron entry name` maybe?

File: lib/chef/resource/cron_d.rb [hunk 2]
```diff
@@ -97,31 +97,31 @@ def self.validate_dow(spec)
...
       property :minute, [Integer, String],
-               description: "The minute to schedule the cron job to run at. Valid values: 0-59.",
+               description: "The minute at which the cron entry should run (0 - 59).",
                default: "*", callbacks: {
                  "should be a valid minute spec" => ->(spec) { validate_numeric(spec, 0, 59) },
                }
 
       property :hour, [Integer, String],
-               description: "The hour to schedule the cron job to run at. Valid values: 0-23.",
+               description: "The hour at which the cron entry is to run (0 - 23).",
```
Comment (tyler-ball) [inline lib/chef/resource/cron_d.rb]: All the other descriptions use `cron entry should run`

File: lib/chef/resource/execute.rb [hunk 3]
```diff
@@ -65,7 +65,7 @@ def command(arg = nil)
                description: "The current working directory from which the command will be run."
 
       property :environment, Hash,
-               description: "Specify a Hash of environment variables to be set."
+               description: "A Hash of environment variables in the form of ({'ENV_VARIABLE' => 'VALUE'}). (These variables must exist for a command to be run successfully.)"
```
Comment (tyler-ball) [inline lib/chef/resource/execute.rb]: The phrasing on this makes it seem a little like this is a required property. What about `If specified, these variables must exist...`

File: lib/chef/resource/hostname.rb [hunk 4]
```diff
@@ -27,7 +27,7 @@ class Hostname < Chef::Resource
       introduced "14.0"
 
       property :hostname, String,
-               description: "Used to specify the hostname if it is different than the resource's name.",
+               description: "An optional property to set the hostnam if it differs from the resource block's name.",
```
Comment (tyler-ball) [inline lib/chef/resource/hostname.rb]: `hostnam` -> `hostname`

File: lib/chef/resource/dmg_package.rb [hunk 5]
```diff
@@ -27,31 +27,31 @@ class DmgPackage < Chef::Resource
       introduced "14.0"
 
       property :app, String,
-               description: "The name of the application used by default for the /Volumes directory and the .app directory copied to /Applications.",
+               description: "The name of the application as it appears in the /Volumes directory, if it differs from the resource block name.",
```
Comment (tyler-ball) [inline lib/chef/resource/dmg_package.rb]: `resource block's name`

File: lib/chef/resource/cron_d.rb (hunk 1 above)
Comment (tas50) [reply lib/chef/resource/cron_d.rb]: These aren't actually entries. They're just standalone job definitions in cron.d

File: lib/chef/resource/execute.rb (hunk 3 above)
Comment (tas50) [reply lib/chef/resource/execute.rb]: That is super odd. This is from our docs site, but it really doesn't make sense.

File: lib/chef/resource/cron_d.rb (hunk 2 above)
Comment (tas50) [reply lib/chef/resource/cron_d.rb]: Yep. Fixed

File: lib/chef/resource/hostname.rb (hunk 4 above)
Comment (tas50) [reply lib/chef/resource/hostname.rb]: Fixed

**Pull request merged**